contributors/design-proposals/raw-block-pv.md

```diff
@@ -0,0 +1,572 @@
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: my-db
+spec:
+    containers
+    - name: mysql
+      image: mysql
+      volumeDevices: #proposed API change
+      - name: my-db-data
+        devicePath: /dev/xvda #proposed API change
+    volumes:
+    - name: my-db-data
+      persistentVolumeClaim:
+	claimName: raw-pvc
+```
+## Storage Class non-API Changes:
+For dynamic provisioning, it is assumed that values passed in the parameter section are opaque, thus the introduction of utilizing
+fsType in the StorageClass can be used by the provisioner to indicate how to create the volume. The proposal for this value is
+defined here:
+https://github.com/kubernetes/kubernetes/pull/45345 
+This section is provided as a general guideline, but each provisioner may implement their parameters independent of what is defined
+here. It is our recommendation that the volumeMode in the PVC be the guidance for the provisioner and overrides the value given in the fstype. Therefore a provisioner should be able to ignore the fstype and provision a block device if that is what the user requested via the PVC and the provisioner can support this.
+
+```
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: block-volume
+ provisioner: kubernetes.io/scaleio
+ parameters:
+  gateway: https://192.168.99.200:443/api
+  system: scaleio
+  protectionDomain: default
+  storagePool: default
+  storageMode: ThinProvisionned
+  secretRef: sio-secret
+  readOnly: false
+  fsType: Block #suggested value
+```
+The provisioner (if applicable) should validate the parameters and return and error if the combination specified is not supported.
+This also allows the use case for leveraging a Storage Class for utilizing pre-defined static volumes. By labeling the Persistent Volumes
+with the Storage Class, volumes can be grouped and used according to how they are defined in the class.
+```
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: block-volume
+provisioner: no-provisioning 
+parameters:
+```
+
+# Pod Security Policy (PSP) Changes:
+Since the utilization of block devices can pose a risk to possible kernel manipulation by malicious users, it may be desirable for the administrator to restrict the usage entirely within a cluster. A similar convention is used with hostPath in that for some kubernetes 
+implmentations it is disabled by default.
+Thus, the PSP can define whether a validating pod can request the usage of such devices through the volumeDevices parameter.
+
+```
+NAME               PRIV      CAPS      SELINUX     RUNASUSER          FSGROUP     SUPGROUP    PRIORITY   READONLYROOTFS   VOLUMES
+anyuid             false     []        MustRunAs   RunAsAny           RunAsAny    RunAsAny    10         false            [configMap downwardAPI emptyDir persistentVolumeClaim secret volumeDevices]
+```
+
+# Use Cases
+
+## UC1: 
```

Comment:
Can you add an example of a pod that uses both filesystem and block volumes?

Reply:
@msau42 UC8 added